### PR TITLE
Added Mino Hotswap PCB

### PR DESCRIPTION
**Pull request by CheeseL0ver**
## Description
Add QMK support for Mino Hotswap PCB with default and VIA keymaps. 

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X] Keyboard (addition or update)
- [X] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [X] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [X] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/mino/hotswap/config.h [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+#define DEBOUNCE 5
+
+/* Mechanical locking support. Use KC_LCAP, KC_LNUM or KC_LSCR instead in keymap */
+#define LOCKING_SUPPORT_ENABLE
+/* Locking resynchronize hack */
+#define LOCKING_RESYNC_ENABLE
+
+/* disable these deprecated features by default */
+#define NO_ACTION_MACRO
+#define NO_ACTION_FUNCTION
```
Comment (drashna) [inline keyboards/mino/hotswap/config.h:60]: deprecated 
```suggestion
```

File: keyboards/mino/hotswap/info.json [hunk 2]
```diff
@@ -0,0 +1,61 @@
+{
+    "keyboard_name": "Mino",
+    "url": "https://qmk.fm/keyboards/",
+    "maintainer": "ShandonCodes",
+    "width": 12,
+    "height": 4,
```
Comment (drashna) [inline keyboards/mino/hotswap/info.json:6]:
```suggestion
```

File: keyboards/mino/hotswap/rules.mk [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+# Bootloader selection
+BOOTLOADER = caterina #Make sure you change this if you are not using the kit Pro Micro
+
+# Build Options
+BOOTMAGIC_ENABLE = no     # Virtual DIP switch configuration
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+NKRO_ENABLE = no            # USB Nkey Rollover
```
Comment (drashna) [inline keyboards/mino/hotswap/rules.mk:12]:
```suggestion
NKRO_ENABLE = no            # Enable N-Key Rollover
```

File: keyboards/mino/hotswap/rules.mk [hunk 4]
```diff
@@ -0,0 +1,14 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = caterina #Make sure you change this if you are not using the kit Pro Micro
+
+# Build Options
+BOOTMAGIC_ENABLE = no     # Virtual DIP switch configuration
```
Comment (drashna) [inline keyboards/mino/hotswap/rules.mk:8]:
```suggestion
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
```

File: keyboards/mino/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,293 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "mino.h"
+#include <stdio.h>
```
Comment (drashna) [inline keyboards/mino/keymaps/default/keymap.c:18]:
```suggestion
#include QMK_KEYBOARD_H
```

File: keyboards/mino/keymaps/via/keymap.c [hunk 6]
```diff
@@ -0,0 +1,301 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "mino.h"
+#include <stdio.h>
```
Comment (drashna) [inline keyboards/mino/keymaps/via/keymap.c:17]:
```suggestion
#include QMK_KEYBOARD_H
```

File: keyboards/mino/readme.md [hunk 7]
```diff
@@ -0,0 +1,19 @@
+# mino
+
+![Mino](https://i.imgur.com/f5kHu8Q.jpg)
```
Comment (drashna) [inline keyboards/mino/readme.md:3]:
```suggestion
![Mino](https://i.imgur.com/f5kHu8Qh.jpg)
```

File: keyboards/mino/readme.md [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+
+A 47 key board with a retro twist.
+
+* Keyboard Maintainer: [ShandonCodes](https://github.com/ShandonCodes)
+* Hardware Supported: Mino PCB
+* Hardware Availability: Groupbuy
+
+Make example for this keyboard (after setting up your build environment):
```
Comment (drashna) [inline keyboards/mino/readme.md:11]: Could you add instructions on how to enter the bootloader, to flash new firmware? 

Eg: 
```markup
## Bootloader

Enter the bootloader in 3 ways:

* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (usually the top left key or Escape) and plug in the keyboard
* **Physical reset button**: Briefly press the button on the back of the PCB - some may have pads you must short instead
* **Keycode in layout**: Press the key mapped to `RESET` if it is available
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/mino/keymaps/default/keymap.c [hunk 9]
```diff
@@ -0,0 +1,294 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+// #include "mino.h"
+// #include <stdio.h>
```
Comment (drashna) [inline keyboards/mino/keymaps/default/keymap.c:18]:
```suggestion
```

File: keyboards/mino/keymaps/default/keymap.c [hunk 10]
```diff
@@ -0,0 +1,294 @@
...
+    [_2] = LAYOUT_default(
+        KC_UP, KC_EXLM, KC_AT, KC_HASH, KC_DLR, KC_PERC, KC_CIRC, KC_AMPR, KC_ASTR, KC_LPRN, KC_RPRN, XXXXXXX, XXXXXXX,
+        KC_LEFT, KC_DOWN, KC_RGHT, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,
+        XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,
+        XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, RESET, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX
+    )
+};
+
+#ifdef OLED_ENABLE
+char wpm_str[10];
```
Comment (drashna) [inline keyboards/mino/keymaps/default/keymap.c:51]:
```suggestion
```

File: keyboards/mino/keymaps/via/keymap.c [hunk 11]
```diff
@@ -0,0 +1,300 @@
...
+    [_3] = LAYOUT_default(
+        XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,
+        XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,
+        XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,
+        XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX
+    )
+};
+
+#ifdef OLED_ENABLE
+char wpm_str[10];
```
Comment (drashna) [inline keyboards/mino/keymaps/via/keymap.c:54]:
```suggestion
```

File: keyboards/mino/mino.h [hunk 12]
```diff
@@ -0,0 +1,21 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+ #pragma once
+
+#include "quantum.h"
+
+#include "hotswap.h"
```
Comment (drashna) [inline keyboards/mino/mino.h:21]:
```suggestion
#ifdef KEYBOARD_mino_hotswap
#    include "hotswap.h"
#endif
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/mino/hotswap/rules.mk [hunk 13]
```diff
@@ -0,0 +1,14 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = caterina #Make sure you change this if you are not using the kit Pro Micro
```
Comment (fauxpark) [inline keyboards/mino/hotswap/rules.mk:5]:
```suggestion
BOOTLOADER = caterina
```

File: keyboards/mino/hotswap/rules.mk [hunk 14]
```diff
@@ -0,0 +1,14 @@
...
+
+# Build Options
+BOOTMAGIC_ENABLE = yes     # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+NKRO_ENABLE = no            # Enable N-Key Rollover
+OLED_ENABLE = yes
+WPM_ENABLE = yes
```
Comment (fauxpark) [inline keyboards/mino/hotswap/rules.mk:14]:
```suggestion
# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
NKRO_ENABLE = no            # Enable N-Key Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
AUDIO_ENABLE = no           # Audio output
OLED_ENABLE = yes
WPM_ENABLE = yes
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**